### PR TITLE
[4.2] add scheduler to System Quick icons

### PR DESCRIPTION
**Pull request by alikon**
related to  #37449 .

### Summary of Changes
add the option to add Scheduler to System Quick icons

### Testing Instructions
apply patch
edit admin module System Quick icon
set Scheduler Icon to show

### Actual result BEFORE applying this Pull Request
N/A

### Expected result AFTER applying this Pull Request
image

### CODE REVIEW

**Review by laoneo — APPROVED** [review]
First I wanted to ask why making another option, but then I saw all the other icons are options as well. I really don't like these hard wires between extensions and on some point we need to make plugins out of them as there is an event for this already. But approving it for now.

**Comment by brianteeman** [comment]
There is a PR that adds the ability to add more icons etc. Probably better to update that one.

**Comment by simbus82** [comment]
> First I wanted to ask why making another option, but then I saw all the other icons are options as well. I really don't like these hard wires between extensions and on some point we need to make plugins out of them as there is an event for this already. But approving it for now.

I also don't like a lot this UI from UX perspective. But at the moment the @alikon solution follows this track.

Probably there should be some multiple selector or/with repeatable fields that gives us the maximum freedom to manage what icons we want to show, proposing lists of extensions, modules and plugins active in the cms. Not a fixed and predetermined (by who?) list of "show / hide" selectors.

File: administrator/modules/mod_quickicon/src/Helper/QuickIconHelper.php [hunk 1]
```diff
@@ -275,6 +275,17 @@ public function getButtons(Registry $params, CMSApplication $application = null)
 				];
 			}
 
+			if ($params->get('show_scheduler'))
+			{
+				$this->buttons[$key][] = [
+					'image'  => 'icon-clock',
+					'link'   => Route::_('index.php?option=com_scheduler'),
```
Comment (simbus82) [inline administrator/modules/mod_quickicon/src/Helper/QuickIconHelper.php]: Maybe we can add the "Link Add" feature, to complete the Quick Icon features

```
'linkadd' => Route::_('index.php?option=com_scheduler&view=select'),
```

**Comment by brianteeman** [comment]
Please see #32671

File: administrator/modules/mod_quickicon/src/Helper/QuickIconHelper.php (hunk 1 above)
Comment (alikon) [reply administrator/modules/mod_quickicon/src/Helper/QuickIconHelper.php]: added the Add task

**Comment by alikon** [comment]
folks the goal of this pr is not to change the current quick icons behaviuor, but to have a 1 click less  way to go to the scheduler

File: administrator/language/en-GB/mod_quickicon.ini [hunk 2]
```diff
@@ -27,6 +27,8 @@ MOD_QUICKICON_MODULE_MANAGER="Modules"
 MOD_QUICKICON_MODULE_MANAGER_ADD="Add Site Module"
 MOD_QUICKICON_NAV_LABEL="Quick Links"
 MOD_QUICKICON_PLUGIN_MANAGER="Plugins"
+MOD_QUICKICON_SCHEDULER="Scheduler"
+MOD_QUICKICON_SCHEDULER_ADD="Add task"
```
Comment (Quy) [inline administrator/language/en-GB/mod_quickicon.ini]:
```suggestion
MOD_QUICKICON_SCHEDULER_ADD="Add Task"
```

**Comment by brianteeman** [comment]
> folks the goal of this pr is not to change the current quick icons behaviuor, but to have a 1 click less way to go to the scheduler

so do it for everything not just this one

**Comment by alikon** [comment]
it's out of the scope of this pr

**Comment by simbus82** [comment]
I have tested this item :white_check_mark: successfully on 19a430ffd379716ba52cd807b67735dd3f288e98<br /><br />

image

Tested on Joomla 4.1.2<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37513">issues.joomla.org/tracker/joomla-cms/37513</a>.</sub>

**Comment by bikashb98** [comment]
I have tested this item :white_check_mark: successfully on 19a430ffd379716ba52cd807b67735dd3f288e98<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37513">issues.joomla.org/tracker/joomla-cms/37513</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37513">issues.joomla.org/tracker/joomla-cms/37513</a>.</sub>

**Comment by brianteeman** [comment]
I am not in favour of this change. Instead #32671 should be used - do it once and do it right

**Comment by richard67** [comment]
> I am not in favour of this change. Instead #32671 should be used - do it once and do it right

It seems that PR is not ready anymore since it has conflicts.

**Comment by brianteeman** [comment]
@richard67 it is still the right way forward

**Comment by richard67** [comment]
> @richard67 it is still the right way forward

But it doesn’t add anything for the scheduler, does it?

**Comment by brianteeman** [comment]
only because the scheduler didnt exist at the time.

my point is that we should do this once for everything

**Comment by brianteeman** [comment]
i will spend some time in the weekend to update that pr

**Comment by chmst** [comment]
I had it on my to do list but I think the quickicon module should be re-worked completely, enabling individual sorting and own classes for icons and better params. I see this PR as a quick  improvement promoting the scheduled task feature.

**Comment by laoneo** [comment]
As this is a new feature it should be added to 4.2 anyway. @alikon can you please rebase the branch. I would not invest more time into it, when other pr's will come up with an improved quickicon architecture, then we should go that way then. On a long term adding more and more options to the module is not sustainable.

**Comment by laoneo** [comment]
Pack to pending<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37513">issues.joomla.org/tracker/joomla-cms/37513</a>.</sub>

**Comment by richard67** [comment]
@alikon Hmm, what you are doing now after the rebase, deleting and creating again files, will mess up the history of these files when your PR will be merged one day. So maybe it would be better just to re-do it with a new PR?

Besides this, @brianteeman wrote that he wants to come up with a better solution this weekend. Maybe wait for that so you don't waste more time here?

**Comment by alikon** [comment]
i've made mistakes in rebase...anyway
let's close then

**Comment by richard67** [comment]
Rebase is not easy when the 4.2-dev branch is behind the 4.1-dev branch regarding commits and a PR was created after the point of time when the 4.2-dev branch was up to date with the 4.1-dev commits so that the PR after the rebase is in advance to the 4.2-dev branch with these commits. Then GitHub shows not only the changes from that PR but also those from the 4.1-dev branch which have not been merged up into 4.2-dev yet.

The easiest way to avoid is is to wait with a rebase of the PR until the next upmerge from 4.1-dev to 4.2-dev has happened.

**Comment by brianteeman** [comment]
My gut feeling is that the correct way forward is to make this a quickicon plugin and then convert the existing module to also be individual plugins and then add new plugins for the missinf quickicons

**Pull request closed without merging**